### PR TITLE
BE - Orders #11

### PR DESCRIPTION
**Pull request by ViktorPelepiak**

### CODE REVIEW

File: src/main/java/com/chnu/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public GenericResponse<Order> update(@RequestBody Order object) {
+        if (object != null) {
+            return GenericResponse.of(orderService.update(object));
+        }
+        return GenericResponse.error("Order is null.");
+    }
+
+    @DeleteMapping("")
+    public GenericResponse<Void> delete(@RequestBody Order object) {
+        orderService.delete(object);
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/controller/OrderController.java]: Do we need this method? It's be easier to use the next one.

File: src/main/java/com/chnu/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table(name = "orders")
+public class Order {
+
+    @Id
+    @Column(name = "order_id")
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/model/Order.java:15]: We need a generation here

File: src/main/java/com/chnu/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    private Long orderId;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "customer_id", referencedColumnName = "user_id", nullable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/model/Order.java]: May this delete a User while deleting an Order?

File: src/main/java/com/chnu/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    private User customer;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "company_id", referencedColumnName = "company_id", nullable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/model/Order.java]: May this delete a Company while deleting an Order?

File: src/main/java/com/chnu/model/Order.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+    @Column(name = "datetime_to")
+    private Date datetimeTo;
+
+    @Column(name = "status")
+    private Status status;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "courier_id", referencedColumnName = "user_id", nullable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/model/Order.java]: May this delete a Courier while deleting an Order?

File: src/main/java/com/chnu/service/impl/OrderService.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+@Service
+@Transactional
+public class OrderService implements IOrderService {
+    private final IOrderRepository orderRepository;
+
+    @Autowired
+    public OrderService(IOrderRepository orderRepository) {
+        this.orderRepository = orderRepository;
+    }
+
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/service/impl/OrderService.java:22]: Add @Override annotation on the methods

**Review by MikeMalyar — CHANGES_REQUESTED** [review]

File: src/main/java/com/chnu/model/Order.java (hunk 3 above)
Comment (MikeMalyar) [reply src/main/java/com/chnu/model/Order.java]: Think this @OnDelete annotations should be removed from here

**Review by MikeMalyar — APPROVED** [review]

**Review by lenyuk1999oo — APPROVED** [review]

**Pull request merged**